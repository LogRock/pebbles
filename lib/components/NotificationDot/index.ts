export { default } from "./NotificationDot";

export type { NotificationDotProps } from "./NotificationDot";
