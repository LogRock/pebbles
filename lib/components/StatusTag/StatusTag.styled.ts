import styled from "styled-components";

import { OverlineLarge } from "../Typography";
import { StatusTagProps } from "./StatusTag.types";

export const StatusTagWrapper = styled.div<
  Pick<StatusTagProps, "variant" | "size">
>`
  display: flex;
  align-items: center;
  justify-content: center;
  width: max-content;
  height: ${({ theme, size }) => theme.statusTag.tagSize[size]};
  padding: 0 ${({ theme }) => theme.spacings.md};

  border: 1px solid
    ${({ theme, variant }) => theme.statusTag.borderColor[variant]};
  border-radius: 64px;

  background-color: ${({ theme, variant }) =>
    theme.statusTag.backgroundColor[variant]};
`;

export const IconWrapper = styled.div<
  Pick<StatusTagProps, "variant" | "label">
>`
  margin-right: ${({ theme, label }) => (label ? theme.spacings.sm : 0)};
  margin-bottom: -2px;

  color: ${({ theme, variant }) => theme.statusTag.iconColor[variant]};
`;

export const StyledOverlineLarge = styled(OverlineLarge)<
  Pick<StatusTagProps, "variant" | "size">
>`
  color: ${({ theme, variant }) => theme.statusTag.labelColor[variant]};
  font-size: ${({ theme, size }) => theme.statusTag.labelSize[size]};
`;
