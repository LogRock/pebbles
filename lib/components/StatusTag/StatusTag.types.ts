import React from "react";

export interface StatusTagProps {
  label?: string;
  size: "micro" | "small" | "medium" | "large";
  variant: "error" | "warning" | "success" | "neutral";
  icon?: React.ReactNode;
  collapsed?: boolean | false;
}
