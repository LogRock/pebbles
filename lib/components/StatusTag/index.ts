export { default } from "./StatusTag";
export { STATUS_TAG_SIZES, STATUS_TAG_VARIANTS } from "./StatusTag.types";
export type { StatusTagProps } from "./StatusTag.types";
