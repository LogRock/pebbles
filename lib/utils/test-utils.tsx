/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
