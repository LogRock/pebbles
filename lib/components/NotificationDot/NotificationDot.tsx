import styled, { css } from "styled-components";
import { STATUS_TAG_VARIANTS } from "../StatusTag";

export interface NotificationDotProps {
  variant: STATUS_TAG_VARIANTS;
}

const NotificationDot = styled.div<NotificationDotProps>`
  ${({ theme, variant = "NEUTRAL_ALT" }) => css`
    display: inline-flex;
    box-sizing: border-box;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: center;
    min-width: ${theme.spacings.md};
    height: ${theme.spacings.md};
    padding: 0 ${theme.spacings.xxsm};

    border-radius: ${theme.spacings.md};

    background: ${theme.statusTag.color.background[variant]};

    color: ${theme.statusTag.color.content[variant]};
    font-family: ${theme.statusTag.typography.SMALL.fontFamily};
    font-size: 10px;
    font-weight: ${theme.statusTag.typography.SMALL.weight};
    line-height: ${theme.statusTag.typography.SMALL.lineHeight};
    text-transform: ${theme.statusTag.typography.SMALL.textTransform};
    vertical-align: super;
  `}
`;

export default NotificationDot;
