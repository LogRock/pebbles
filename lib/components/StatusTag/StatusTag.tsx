import React, { FC } from "react";

import Icon from "@mdi/react";
import {
  mdiAlert,
  mdiAlertCircle,
  mdiMinusCircle,
  mdiCheckCircle,
} from "@mdi/js";

import { StatusTagProps } from "./StatusTag.types";
import {
  IconWrapper,
  StatusTagWrapper,
  StyledOverlineLarge,
} from "./StatusTag.styled";

const StatusTag: FC<StatusTagProps> = ({ label, size, variant }) => {
  const getIcon = () => {
    switch (variant) {
      case "error":
        return mdiAlert;
      case "warning":
        return mdiAlertCircle;
      case "success":
        return mdiCheckCircle;
      case "neutral":
        return mdiMinusCircle;
      default:
        return mdiMinusCircle;
    }
  };

  const getIconSize = () => {
    switch (size) {
      case "small":
        return 0.6;
      case "medium":
        return 0.75;
      case "large":
        return 0.85;
      default:
        return 0.75;
    }
  };

  return (
    <StatusTagWrapper variant={variant} size={size}>
      <IconWrapper variant={variant}>
        <Icon path={getIcon()} size={getIconSize()} />
      </IconWrapper>
      <StyledOverlineLarge variant={variant} size={size}>
        {label}
      </StyledOverlineLarge>
    </StatusTagWrapper>
  );
};

export default StatusTag;
