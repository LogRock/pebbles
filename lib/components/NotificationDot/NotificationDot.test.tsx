import React from "react";
import { render } from "../../utils/test-utils";
import { STATUS_TAG_VARIANTS } from "../StatusTag";
import NotificationDot from "./NotificationDot";

describe("NotificationDot", () => {
  it("renders", () => {
    const { container } = render(
      <NotificationDot variant={STATUS_TAG_VARIANTS.NEUTRAL}>
        Foo
      </NotificationDot>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
