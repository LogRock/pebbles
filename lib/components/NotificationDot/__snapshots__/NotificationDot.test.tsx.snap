// Vitest Snapshot v1

exports[`NotificationDot > renders 1`] = `
<div
  class="sc-gEvEer iNyIYE"
>
  Foo
</div>
`;
